app/models/alchemy/page/publisher.rb

```diff
@@ -19,9 +19,15 @@ def publish!(public_on:)
           version = public_version(public_on)
           DeleteElements.new(version.elements).call
 
-          # We must not use .find_each here to not mess up the order of elements
-          page.draft_version.elements.not_nested.available.each do |element|
-            Element.copy(element, page_version_id: version.id)
+          repository = page.draft_version.element_repository
+          ActiveRecord::Base.no_touching do
+            Element.acts_as_list_no_update do
+              repository.visible.not_nested.each do |element|
```

Comment:
Since we skip hidden elements we will have inconsistent positioning. 

What about using `each.with_index(1)` here and pass the index as `position` to the new element. 

This should have no effect on the performance as long as we still keep acts-as-list disabled. 


```suggestion
              repository.visible.not_nested.each.with_index(1) do |element, position|
```